src/qs_environment.F

```diff
@@ -792,6 +793,25 @@ SUBROUTINE qs_init_subsys(qs_env, para_env, subsys, cell, cell_ref, use_ref_cell
          END DO
       END IF
 
+      IF (dft_control%do_xas_tdp_calculation) THEN
+         ! Check if RI_XAS basis is given, auto-generate if not
+         CALL get_qs_env(qs_env, nkind=nkind)
+         DO ikind = 1, nkind
+            NULLIFY (ri_xas_basis)
+            qs_kind => qs_kind_set(ikind)
+            CALL get_qs_kind(qs_kind, basis_Set=ri_xas_basis, basis_type="RI_XAS")
+            IF (.NOT. ASSOCIATED(ri_xas_basis)) THEN
+               CALL cp_warn(__LOCATION__, "Automatic Generation of RI_XAS basis. "// &
+                            "This is experimental code.")
+               !Because might need nco(l) with high l<=max_orbl in create_ri_aux_basis_set
+               CALL get_qs_kind_set(qs_kind_set, maxlgto=maxlgto)
+               CALL init_orbital_pointers(2*maxlgto)
```

Comment:
Do you think this could be taken care of in `create_ri_aux_basis_set`? It's a bit unfortunate that this needs to be done manually before creating an RI basis.

Reply:
Yes indeed, especially since it could happen with the other RI basis sets as well. I'll move that bit of code